### ISSUE TITLE
Multiplicative Complexity

### ISSUE THREAD

**Issue by zfx-code** [peigen-sboxes/PEIGEN#12]
Dear Author
I want to know how to optimaze MC. I mean Multiplicative number(not nonlinear number).
That's I restrict use {XOR, AND, NOT} and the {XOR, NOT} is unlimited.
I only kept one sbox(Piccolo,0e040b0203080009010a070f060c050d) in sboxes4.txt.
I try run 
```sh
./search_GC_n4 -v -c 6 -l 6 -p 4 -r 160 --xor2 --and2 --not1 -f software.conf.
```
And set not1_cost=xor2_cost=0.00, and2_cost=1.00 in software.conf.
But in the pre-computation state, it takes up a lot of memory and disk.
So maybe I have something wrong?

Looking forward to your reply.
Best wishes to you.

**Comment by FreeDisciplina** [comment]
Yes, sorry that the pre-computation requires a lot of memory and disk. The pre-computation is designed for efficiently searching implementations of _many_ S-boxes.
For a _single_ S-box, storing the tree on a disk is not worth it. One can skip the pre-computation, set `-c 0`, and use the function `search_single()` defined in [lighter_mitm.hpp](https://github.com/peigen-sboxes/PEIGEN/blob/master/lighter_mitm.hpp#L650).

**Comment by FreeDisciplina** [comment]
Sorry, I am afraid that one cannot set not1_cost=0.00 and xor2_cost=0.00. If so, I think the program will expand the tree using NOT1 and XOR2 endlessly.
